src/utils/utils.ts

```diff
@@ -0,0 +1,165 @@
+ */
+function findUserESLintConfig(repoRoot = process.cwd()): string | null {
+  const candidates = [
+    'eslint.config.js',
+    'eslint.config.mjs',
+    'eslint.config.cjs',
+    'eslint.config.ts',
+  ];
+  for (const file of candidates) {
+    const abs = path.join(repoRoot, file);
+    if (fs.existsSync(abs)) return abs;
+  }
+  return null;
+}
+
+/**
+ * Collect all Markdown files in a directory and its subdirectories.
+ *
+ * @param dir The directory to search in.
+ * @returns An array of paths to Markdown files.
+ */
+function collectMarkdown(dir: string): string[] {
+  const entries = fs.readdirSync(dir, { withFileTypes: true });
+  const files: string[] = [];
+  for (const e of entries) {
+    const p = path.join(dir, e.name);
+    if (e.isDirectory()) {
+      files.push(...collectMarkdown(p));
+    } else if (e.isFile() && /\.(md|mdx)$/i.test(e.name)) {
+      files.push(p);
+    }
+  }
+  return files;
+}
+
+/**
+ * Check if a command exists in the system PATH.
+ *
+ * @param cmd The command to check.
+ * @returns True if the command exists, false otherwise.
+ */
+function commandExists(cmd: string): boolean {
+  const whichCmd = process.platform === 'win32' ? 'where' : 'which';
+  const result = childProcess.spawnSync(whichCmd, [cmd], { stdio: 'ignore' });
+  return result.status === 0;
+}
+
+interface MatrixAILintConfig {
+  tsconfigPaths: string[];
+  forceInclude: string[];
+}
+
+function resolveMatrixConfig(repoRoot = process.cwd()): MatrixAILintConfig {
+  const cfgPath = path.join(repoRoot, 'matrixai-lint-config.json');
+
+  const abs = (p: string) => path.resolve(repoRoot, p);
+  const exists = (p: string) => fs.existsSync(p);
+
+  let rawCfg: unknown = {};
+
+  if (exists(cfgPath)) {
+    try {
+      const text = fs.readFileSync(cfgPath, 'utf8').trim();
+      rawCfg = text.length ? JSON.parse(text) : {};
+    } catch {
+      console.error(
+        '[matrixai‑lint]  ✖  matrixai-lint-config.json is not valid JSON – falling back to defaults.',
+      );
+    }
+  }
+
+  const toStringArray = (v: unknown): string[] =>
+    typeof v === 'string'
+      ? [v]
+      : Array.isArray(v)
+        ? v.filter((x): x is string => typeof x === 'string')
+        : [];
+
+  const cfg = rawCfg as { tsconfigPaths?: unknown; forceInclude?: unknown };
+
+  const tsconfigPaths = toStringArray(cfg.tsconfigPaths)
+    .map(abs)
+    .filter((p) => {
+      if (exists(p)) return true;
+      console.warn(`[matrixai‑lint]  ⚠  tsconfig not found: ${p}`);
+      return false;
+    });
+
+  const forceInclude = toStringArray(cfg.forceInclude).map((g) =>
+    g.replace(/^\.\//, ''),
+  );
+
+  // Fallback to root tsconfig
+  if (tsconfigPaths.length === 0) {
+    const rootTs = path.join(repoRoot, 'tsconfig.json');
+    if (exists(rootTs)) tsconfigPaths.push(rootTs);
+  }
+
+  return { tsconfigPaths, forceInclude };
+}
+
+interface Patterns {
+  files: string[];
+  ignore: string[];
+}
```

Comment:
Don't use interfaces for types, just define a new type. But you don't even need to do that when it's only used once. In that case just define the return type directly like this.

```ts
function buildPatterns(
  tsconfigPath: string,
  forceInclude: string[] = [],
): {
  files: string[];
  ignore: string[];
}
```

If you had to use a type multiple times then define it with 

```ts
type Pattern = {
  files: string[];
  ignore: string[];
};
```

And if this is used across files the define it in a `types.ts` file.